shell/platform/windows/windows_lifecycle_manager.cc

```diff
@@ -0,0 +1,134 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "windows_lifecycle_manager.h"
+
+#include <TlHelp32.h>
+#include <WinUser.h>
+#include <Windows.h>
+#include <tchar.h>
+
+#include "flutter/shell/platform/windows/flutter_windows_engine.h"
+
+namespace flutter {
+
+WindowsLifecycleManager::WindowsLifecycleManager(FlutterWindowsEngine* engine)
+    : engine_(engine) {}
+
+WindowsLifecycleManager::~WindowsLifecycleManager() {}
+
+void WindowsLifecycleManager::Quit(UINT exit_code, HWND hwnd) {
+  if (hwnd == nullptr) {
+    ::PostQuitMessage(exit_code);
+  } else {
+    sent_close_messages_[hwnd]++;
+    PostMessage(hwnd, WM_CLOSE, 0, 0);
```

Comment:
We should maintain the `wparam` and `lparam` values when we redispatch (you can do evil tricks and shove state in these unused values). Here's how the keyboard does it:

https://github.com/flutter/engine/blob/252f6cfd0335918d4147ba63e52b5800c7254478/shell/platform/windows/keyboard_manager.cc#L118-L120